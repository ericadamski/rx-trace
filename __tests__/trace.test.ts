import { from, of } from 'rxjs';
import { tap } from 'rxjs/operators';
import trace from '../index';

const mockOf = of;

describe('trace', () => {
  const label = 'label';
  const arr = [1];

  it('should log output with the given label', () => {
    const log = jest.fn();

    return from(arr)
      .pipe(trace(label, log))
      .toPromise()
      .then(() => {
        expect(log).toHaveBeenCalledTimes(arr.length);
        // @ts-ignore
        expect(log.mock.calls).toMatchSnapshot();
      });
  });

  it('should call rx-js tap operator', () => {
    // @ts-ignore
    tap = jest.fn(() => () => mockOf({}));

    return from(arr)
      .pipe(trace(label))
      .toPromise()
      .then(() => {
        expect(tap).toHaveBeenCalledTimes(1);
      });
  });
});
