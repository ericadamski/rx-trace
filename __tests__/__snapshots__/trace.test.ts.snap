// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`trace should log output with the given label 1`] = `
Array [
  Array [
    "label",
    1,
  ],
]
`;
